scipy/stats/stats.py

```diff
@@ -105,6 +105,8 @@
    spearmanr
    pointbiserialr
    kendalltau
+   weightedrankedtau
+   weightedtau
```

Comment:
These two functions are so similar that they should be merged. I'd use ``weightedtau`` as the name, and add ``rank=None, use_ranked=False`` keywords (or something similar).